src/test/java/com/xceptance/neodymium/util/SelenideAddonsTest.java

```diff
@@ -640,4 +642,33 @@ public void testOpenHtmlContentWithCurrentWebDriver()
         SelenideAddons.openHtmlContentWithCurrentWebDriver(textHtml);
         Assert.assertEquals(text, $("body").getText());
     }
+
+    @Test
+    public void testOptionalWaitUntil()
+    {
+        openBlogPage();
+        SelenideElement privaceDialog = $("#privacy-message");
+        boolean isVisible = SelenideAddons.optionalWaitUntilCondition(privaceDialog, visible);
+        assertTrue(isVisible);
+        long startTime = new Date().getTime();
+        boolean isHidden = SelenideAddons.optionalWaitUntilCondition(privaceDialog, hidden);
+        long endTime = new Date().getTime();
+        assertFalse(isHidden);
+        Assert.assertTrue(endTime - startTime > Neodymium.configuration().optionalElementRetryTimeout());
```

Comment:
If we know, that the ` SelenideAddons.optionalWaitUntilCondition(privaceDialog, hidden);` should return `false`, then we can expect, that the method should take maximal waiting time -> `Neodymium.configuration().optionalElementRetryTimeout() * Neodymium.configuration().optionalElementRetryCount();`. This means, we can improve assert with following code:
```suggestion
int waitingTime = Neodymium.configuration().optionalElementRetryTimeout() * Neodymium.configuration().optionalElementRetryCount();
        Assert.assertTrue(Range.between(waitingTime, waitingTime + Neodymium.configuration().optionalElementRetryTimeout()).contains(Math.toIntExact(endTime - startTime)));
```

where Range comes from `org.apache.commons.lang3` package.
What do you think?

Reply:
done